tests/test_simple_app.py

```
@@ -87,23 +87,21 @@ async def serve():
     assert "hello" in response.headers["x-custom-header"]
 
 
-def test_empty_file(capsys):
-    captured = capsys.readouterr()
-    with pytest.raises(SystemExit) as e:
-        app = Stardust().build()
-        assert captured.out == "File must contain at least one local function."
+def test_empty_file():
+    with pytest.raises(InvalidApplicationError, match="File must contain at least one local function."):
```

Ruff (0.15.4)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)